.github/workflows/precommit.yml

```diff
@@ -246,6 +246,62 @@ jobs:
             .tox/.coverage
             .tox/coverage.xml
 
+  test-ast-encoding:
+    name: Test AST Encoding py-${{ matrix.os }}-${{ matrix.python-version }}
+    needs: build
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, windows-latest, ubuntu-latest]
+        # TODO SNOW-1763185 use py39 by default
+        # Protobuf encoding differs per python version, use 3.8 by default.
+        python-version: [ "3.8"]
+        cloud-provider: [aws]
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Display Python version
+        run: python -c "import sys; print(sys.version)"
+      - name: Decrypt parameters.py
+        shell: bash
+        run: .github/scripts/decrypt_parameters.sh
+        env:
+          PARAMETER_PASSWORD: ${{ secrets.PARAMETER_PASSWORD }}
+          CLOUD_PROVIDER: ${{ matrix.cloud-provider }}
+      - name: Download wheel(s)
+        uses: actions/download-artifact@v4
+        with:
+          name: wheel
+          path: dist
+      - name: Show wheels downloaded
+        run: ls -lh dist
+        shell: bash
+      - name: Upgrade setuptools, pip and wheel
+        run: python -m pip install -U setuptools pip wheel
+      - name: Install tox
+        run: python -m pip install tox
+      - name: Set up windows environment
+        if: runner.os == 'Windows'
+        # TODO: SNOW-1763186 use pytest marker to test multiple TZs
+        run: |
+            "TZ=EST" >> $env:GITHUB_ENV
```

Comment:
This was an oversight on my part when I reviewed the initial changes. I think we should avoid time zones such as `EST`/`EDT`, and we should use IANA time zone names instead. Here, and one line below, the correct name would be `America/New_York`.

@sfc-gh-lspiegelberg FYI.

Reply:
If only Windows understood them in their environment variable... 😬😬😬😬😬😬

Reply:
Sigh... Please disregard, @sfc-gh-azhan  😢

Reply:
I added a todo there to resolve tz after we have all pieces in.